### ISSUE TITLE
Autocomplete not working in certain files, possible 0.28.0 regression

### ISSUE THREAD

**Issue by sam007961** [microsoft/vscode-cpptools#5481]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Fedora 31
- VS Code Version: 1.45.0
- C/C++ Extension Version: 0.28.0
- Other extensions you installed (and if the issue persists after disabling them): CMake Tools 1.3.1, Persists after disabling
- Does this issue involve using SSH remote to run the extension on a remote machine?: No
- Autocomplete is not working in certain files after updating to 0.28.0, although it works fine in the same files using 0.27.1. However, hovering over symbols to see their definitions still works.

**Steps to reproduce**

1. Clone this repo: https://github.com/sam007961/toy-browser
2. Open the file `src/style.cpp` or the file `test/style-test.cpp`
3. Failed to read response from server error shows up in logs.
4. Type `style::` inside any function.
5. No autocomplete suggestions show up.

**Expected behavior**

The contents of the `style` namespace should show up in the suggestions.

<details>
  <summary><strong>Logs</strong></summary>

```
-------- Diagnostics - 5/13/2020, 4:08:49 PM
Version: 0.28.0
Current Configuration:
{
    "name": "Linux",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [],
    "compilerPath": "/opt/cling/bin/clang",
    "cStandard": "c11",
    "cppStandard": "c++17",
    "intelliSenseMode": "clang-x64",
    "compilerArgs": [],
    "configurationProvider": "vector-of-bool.cmake-tools",
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ /home/sam/Projects/toy-browser/src/style.cpp ]:
    /home/sam/Projects/toy-browser/src/style.cpp
[ /home/sam/Projects/toy-browser/src/layout.cpp ]:
    /home/sam/Projects/toy-browser/src/layout.cpp
[ /home/sam/Projects/toy-browser/test/style-test.cpp ]:
    /home/sam/Projects/toy-browser/test/style-test.cpp
Translation Unit Configurations:
[ /home/sam/Projects/toy-browser/src/style.cpp ]: not ready
    Process ID: 20103
    Memory Usage: 0 MB
    Compiler Path: /usr/bin/g++
    Includes:
        /home/sam/Projects/toy-browser/src
        /usr/include/c++/9
        /usr/include/c++/9/x86_64-redhat-linux
        /usr/include/c++/9/backward
        /usr/lib/gcc/x86_64-redhat-linux/9/include
        /usr/local/include
        /usr/include
    Standard Version: c++17
    IntelliSense Mode: gcc-x64
    Other Flags:
        --g++
        --gnu_version=90301
[ /home/sam/Projects/toy-browser/src/layout.cpp ]:
    Process ID: 20139
    Memory Usage: 65 MB
    Compiler Path: /usr/bin/g++
    Includes:
        /home/sam/Projects/toy-browser/src
        /usr/include/c++/9
        /usr/include/c++/9/x86_64-redhat-linux
        /usr/include/c++/9/backward
        /usr/lib/gcc/x86_64-redhat-linux/9/include
        /usr/local/include
        /usr/include
    Standard Version: c++17
    IntelliSense Mode: gcc-x64
    Other Flags:
        --g++
        --gnu_version=90301
[ /home/sam/Projects/toy-browser/test/style-test.cpp ]: not ready
    Process ID: 20158
    Memory Usage: 0 MB
    Compiler Path: /usr/bin/g++
    Includes:
        /home/sam/Projects/toy-browser/src
        /home/sam/Projects/toy-browser/build/googletest-src/googletest/include
        /home/sam/Projects/toy-browser/build/googletest-src/googletest
        /usr/include/c++/9
        /usr/include/c++/9/x86_64-redhat-linux
        /usr/include/c++/9/backward
        /usr/lib/gcc/x86_64-redhat-linux/9/include
        /usr/local/include
        /usr/include
    Standard Version: c++17
    IntelliSense Mode: gcc-x64
    Other Flags:
        --g++
        --gnu_version=90301
Total Memory Usage: 65 MB
```

```
Failed to read response from server: 20
Failed to read response from server: 22
```
</details>

**Screenshots**

**Additional context**

**Comment by sean-mcmanus** [comment]
I got a crash repro on Linux.

**Comment by bobbrow** [comment]
```
(lldb) bt
* thread #1, name = 'cpptools-srv', stop reason = signal SIGSEGV
  * frame #0: 0x000000000090b2c5 cpptools-srv`copy_template_arg_list_with_substitution(a_symbol*, a_template_arg*, a_template_param*, a_template_param*, a_template_arg*, a_template_param*, a_source_position*, int,
 int*, a_ctws_state*) + 309
    frame #1: 0x000000000090ce37 cpptools-srv`copy_template_class_reference_with_substitution(a_symbol*, a_type*, a_template_arg*, a_template_param*, a_source_position*, int, int*, a_ctws_state*, a_type**) + 455
    frame #2: 0x000000000090cc21 cpptools-srv`copy_parent_type_with_substitution(a_symbol*, a_type*, a_template_arg*, a_template_param*, a_source_position*, int, int, int*, a_ctws_state*) + 1281
    frame #3: 0x000000000090750e cpptools-srv`copy_type_with_substitution(a_type*, a_template_arg*, a_template_param*, a_source_position*, int, int*, a_ctws_state*) + 382
    frame #4: 0x000000000090bf66 cpptools-srv`substitute_template_argument(a_template_arg*, a_template_param*, a_template_arg*, a_template_param*, a_template_arg*, a_template_param*, a_source_position*, int, int,
int*, a_ctws_state*) + 150
    frame #5: 0x000000000090b7a1 cpptools-srv`copy_template_arg_list_with_substitution(a_symbol*, a_template_arg*, a_template_param*, a_template_param*, a_template_arg*, a_template_param*, a_source_position*, int,
 int*, a_ctws_state*) + 1553
    frame #6: 0x000000000075ce66 cpptools-srv`rescan_expr_with_substitution_internal(an_expr_node*, a_rescan_control_block*, int, an_operand*, an_operand*) + 998
    frame #7: 0x00000000007787c5 cpptools-srv`rescan_expr_with_substitution(an_expr_node*, a_type*, a_rescan_control_block*, a_constant*) + 325
    frame #8: 0x00000000007dcc32 cpptools-srv`copy_template_param_expr(an_expr_node*, a_template_arg*, a_template_param*, a_type*, a_source_position*, int, int*, a_ctws_state*, a_constant*, a_constant**) + 226
    frame #9: 0x00000000007dfd98 cpptools-srv`copy_template_param_con(a_constant*, a_template_arg*, a_template_param*, a_type*, a_source_position*, int, int*, a_ctws_state*, a_constant*) + 2984
    frame #10: 0x00000000007e0d74 cpptools-srv`copy_template_param_con_with_substitution(a_constant*, a_template_arg*, a_template_param*, a_type*, a_source_position*, int, int*, a_ctws_state*) + 420
    frame #11: 0x00000000007e17a3 cpptools-srv`substitute_constant(a_constant**, a_type*, a_template_param*, a_template_arg*, int, a_ctws_state*, a_source_position*, int*) + 243
    frame #12: 0x00000000007e176f cpptools-srv`substitute_constant(a_constant**, a_type*, a_template_param*, a_template_arg*, int, a_ctws_state*, a_source_position*, int*) + 191
    frame #13: 0x0000000000902422 cpptools-srv`copy_exc_spec_from_prototype_template(an_exception_specification*, int*) + 578
    frame #14: 0x0000000000908503 cpptools-srv`copy_type_with_substitution(a_type*, a_template_arg*, a_template_param*, a_source_position*, int, int*, a_ctws_state*) + 4467
    frame #15: 0x00000000008fef9b cpptools-srv`substitute_template_arguments(a_symbol*, a_template_arg*, a_template_arg**, a_template_param*, int) + 411
    frame #16: 0x00000000008fe6f0 cpptools-srv`wrapup_function_template_argument_deduction(a_template_arg**, a_symbol*, a_template_param*, int, unsigned int) + 144
    frame #17: 0x00000000008b8747 cpptools-srv`determine_function_viability(a_symbol*, a_symbol*, int, a_template_arg*, a_symbol*, a_type*, an_init_component*, int, an_operand*, a_type*, int, int, int, int, int, i
nt, int, int, int, int, an_overload_context, a_candidate_function**, int*, int*, int*, int*) + 3911
    frame #18: 0x0000000000899f18 cpptools-srv`try_overloaded_function_match(a_symbol*, int, a_template_arg*, an_init_component*, an_init_component*, int, an_operand*, int, int, int, int, int, int, int, int, int,
int, an_overload_context, a_candidate_function**, a_symbol**, int*, int*, int*) + 1272
    frame #19: 0x000000000089b07f cpptools-srv`select_overloaded_function(a_symbol*, int, a_template_arg*, int, an_operand*, an_init_component*, an_init_component*, int, int, int, int, an_overload_context, a_sourc
e_position*, unsigned int, int*, int*, int*, int*, a_symbol**, an_arg_match_summary**) + 3055
    frame #20: 0x0000000000740028 cpptools-srv`scan_ctor_arguments(a_symbol*, a_source_position*, a_type*, a_type*, int, int, int, int, a_rescan_control_block*, int, an_init_component*, an_init_component*, int*, i
nt*, int*, int*, int*, an_operand*, a_dynamic_init**, an_expr_node**, a_source_position*) + 1160
    frame #21: 0x00000000008a8bb6 cpptools-srv`prep_list_initializer(an_init_component*, a_type*, int, int, int, int, int, int, int, an_operand*, an_init_state*, an_arg_match_summary*) + 10390
    frame #22: 0x0000000000754335 cpptools-srv`convert_initializer(an_init_component*, a_type*, int, int, an_init_state*) + 357
    frame #23: 0x000000000071238d cpptools-srv`braced_initializer(a_type*, an_init_component*, an_init_state*, a_decl_parse_state*, int, an_init_component**, a_source_position*) + 589
    frame #24: 0x000000000071e62b cpptools-srv`braced_mem_initializer(a_routine*, a_type*, a_constructor_init*, an_init_component*) + 187
    frame #25: 0x000000000071a67d cpptools-srv`scan_mem_init_args(a_routine*, a_constructor_init*, a_type*, a_type*, a_source_position*) + 301
    frame #26: 0x0000000000718d26 cpptools-srv`ctor_initializer(a_routine*, int) + 8870
    frame #27: 0x00000000007c4792 cpptools-srv`scan_function_body(a_routine*, a_func_info_block*, unsigned long, a_macro_arg_fixup**, a_macro_arg_fixup**) + 4722
    frame #28: 0x000000000091ed05 cpptools-srv`instantiate_template_function_full(a_template_instance*, a_routine*, int) + 1173
    frame #29: 0x000000000091c0d9 cpptools-srv`instantiate_entity(a_template_instance*) + 697
    frame #30: 0x000000000090fc87 cpptools-srv`update_instantiation_required_flag(a_template_instance*, int, int) + 1095
    frame #31: 0x0000000000907323 cpptools-srv`set_instance_required(a_symbol*, int, int) + 99
    frame #32: 0x00000000007890e4 cpptools-srv`expr_reference_to_implicitly_invoked_function(a_symbol*, a_source_position*, a_type*, int) + 212
    frame #33: 0x00000000008abd59 cpptools-srv`set_up_for_constructor_call(an_operand*, a_routine*, int, a_conv_descr*, an_expr_node**, int*) + 297
    frame #34: 0x00000000008ad505 cpptools-srv`determine_dynamic_init_for_class_init(an_operand*, a_type*, a_conv_descr*, a_conv_descr*, int, int, int*, a_dynamic_init**, an_expr_node**) + 1941
    frame #35: 0x00000000008ac67a cpptools-srv`prep_elision_initializer_operand(an_operand*, a_type*, int, int, an_error_code, int*, a_dynamic_init**) + 490
    frame #36: 0x00000000008a71b4 cpptools-srv`prep_list_initializer(an_init_component*, a_type*, int, int, int, int, int, int, int, an_operand*, an_init_state*, an_arg_match_summary*) + 3732
    frame #37: 0x0000000000754335 cpptools-srv`convert_initializer(an_init_component*, a_type*, int, int, an_init_state*) + 357
    frame #38: 0x000000000071c82d cpptools-srv`aggr_init_simple_element(an_init_component**, a_type*, an_init_state*, a_constant**) + 1261
    frame #39: 0x000000000071c069 cpptools-srv`aggr_init_element_full(an_init_component**, a_type*, a_field*, an_init_state*, a_source_position*, a_constant**) + 1801
    frame #40: 0x000000000071d5a5 cpptools-srv`aggr_init_field(an_init_component**, a_field**, an_init_state*, a_constant*, a_source_position*) + 389
    frame #41: 0x00000000007114c6 cpptools-srv`aggr_init_class(an_init_component**, a_type*, an_init_state*, a_source_position*, a_constant**) + 2070
    frame #42: 0x000000000070f86c cpptools-srv`prep_aggr_initializer(an_init_component*, a_type**, an_init_state*, an_arg_match_summary*, int) + 828
    frame #43: 0x00000000008a7d89 cpptools-srv`prep_list_initializer(an_init_component*, a_type*, int, int, int, int, int, int, int, an_operand*, an_init_state*, an_arg_match_summary*) + 6761
    frame #44: 0x0000000000784e85 cpptools-srv`scan_braced_init_list_cast(a_type*, a_cast_source_form, an_init_component*, int, an_operand*) + 245
    frame #45: 0x000000000076465f cpptools-srv`scan_functional_notation_type_conversion(a_rescan_control_block*, a_dynamic_init*, int, an_init_component*, a_type*, a_source_position*, an_operand*, int) + 1343
    frame #46: 0x0000000000756f4a cpptools-srv`scan_identifier(an_operand*, int, int, a_rescan_control_block*, a_symbol*, an_operand*, int, a_template_arg*, a_symbol**, int*) + 1882
    frame #47: 0x0000000000745301 cpptools-srv`scan_expr_full(an_operand*, an_operand*, int, int) + 2257
    frame #48: 0x000000000073eee8 cpptools-srv`scan_expr_list(a_token_kind, int, int, int, int, int) + 760
    frame #49: 0x000000000073faab cpptools-srv`scan_call_arguments(a_type*, a_routine*, int, an_expr_node**, int, int, int, int, a_rescan_control_block*, int, an_init_component*, an_init_component**, an_operand*,
int*, a_source_position*) + 779
    frame #50: 0x000000000076ff11 cpptools-srv`scan_function_call(an_operand*, an_operand*, a_rescan_control_block*, an_operand*) + 5409
    frame #51: 0x000000000074709c cpptools-srv`scan_expr_full(an_operand*, an_operand*, int, int) + 9836
    frame #52: 0x0000000000752ba5 cpptools-srv`scan_void_expression(int, int, int, a_dynamic_init**, an_init_component*) + 245
    frame #53: 0x00000000008dd96c cpptools-srv`expression_statement(int) + 124
    frame #54: 0x00000000008dbe45 cpptools-srv`statement(int, int) + 17765
    frame #55: 0x00000000008d6ca9 cpptools-srv`compound_statement_full(int, int, int, int, int, a_type**) + 809
    frame #56: 0x00000000007c47c4 cpptools-srv`scan_function_body(a_routine*, a_func_info_block*, unsigned long, a_macro_arg_fixup**, a_macro_arg_fixup**) + 4772
    frame #57: 0x000000000066f314 cpptools-srv`process_deferred_class_fixups_and_instantiations(int) + 1700
    frame #58: 0x000000000072454d cpptools-srv`class_specifier(a_decl_parse_state*, unsigned long, int, int, int, a_source_position, a_type**, int*, int*, a_decl_pos_block*) + 17069
    frame #59: 0x000000000072acee cpptools-srv`decl_specifiers(unsigned long, a_decl_parse_state*, a_decl_pos_block*) + 1118
    frame #60: 0x000000000070626b cpptools-srv`scan_nonmember_declaration(a_decl_parse_state*, a_source_range*) + 2891
    frame #61: 0x000000000070df6d cpptools-srv`namespace_declaration(a_token_kind*, int, int, a_source_position*, a_symbol**) + 4173
    frame #62: 0x0000000000706492 cpptools-srv`scan_nonmember_declaration(a_decl_parse_state*, a_source_range*) + 3442
    frame #63: 0x000000000070df6d cpptools-srv`namespace_declaration(a_token_kind*, int, int, a_source_position*, a_symbol**) + 4173
    frame #64: 0x0000000000706492 cpptools-srv`scan_nonmember_declaration(a_decl_parse_state*, a_source_range*) + 3442
    frame #65: 0x000000000070bba9 cpptools-srv`declaration(int, int, int, int, a_param_id*, a_source_range*) + 265
    frame #66: 0x0000000000600193 cpptools-srv`compile_fragment((anonymous namespace)::a_fragment_buffer const&, a_canonicalized_file_name, msvc::strong_typedef_t<msvc::offset_traits>, msvc::strong_typedef_t<msvc:
:offset_traits>, msvc::strong_typedef_t<msvc::offset_traits>, bool, a_declaration_context const&, a_region_scope_info const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<cha
r> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, bool, bool, bool, bool*) + 3379
    frame #67: 0x00000000005fd8d2 cpptools-srv`before_tu_wrapup() + 1186
    frame #68: 0x0000000000940bca cpptools-srv`process_translation_unit(char const*, int, an_exported_template_file*) + 1002
    frame #69: 0x000000000066bf87 cpptools-srv`cfe_main(int, char**) + 103
    frame #70: 0x000000000095f916 cpptools-srv`cfe_main_exception_handler(int, char**) + 6
    frame #71: 0x000000000066c096 cpptools-srv`edg_main(int, char**) + 6
    frame #72: 0x000000000095b0d6 cpptools-srv`edge_compiler_main(int, char const**) + 8326
    frame #73: 0x00000000005fdf97 cpptools-srv`preparse(int, char const**, a_scout_store*, edge::translation_unit*) + 215
    frame #74: 0x00000000005c8d2f cpptools-srv`a_compiler_thread::compiler_thread_routine(a_compiler_thread*) + 367
    frame #75: 0x0000000000a574a0 cpptools-srv`execute_native_thread_routine + 16
    frame #76: 0x00007f52d5b8c422 libpthread.so.0`start_thread + 226
    frame #77: 0x00007f52d5abbb83 libc.so.6`__clone + 67
```

**Comment by sean-mcmanus** [comment]
This issue is going to be difficult to get a fix for because it only repros with Release Linux bits, not Debug Linux bits (which is very strange), and not with Mac or Windows.

**Comment by bobbrow** [comment]
The repro is in #5584. Closing this one as a duplicate.

**Comment by sean-mcmanus** [comment]
FYI, this should be fixed in our next 0.29.0-insiders release (not 0.28.3).

**Comment by sean-mcmanus** [comment]
This crash should be fixed with https://github.com/microsoft/vscode-cpptools/releases/tag/0.29.0-insiders .
